pkg/store/mongodb/mongodbStore.go

```diff
@@ -0,0 +1,852 @@
+package mongodb
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/teramoby/speedle-plus/api/pms"
+	"github.com/teramoby/speedle-plus/pkg/errors"
+	"github.com/teramoby/speedle-plus/pkg/suid"
+)
+
+type Store struct {
+	client   *mongo.Client
+	Database string
+}
+
+// ReadPolicyStore reads policy store from a file
+func (s *Store) ReadPolicyStore() (*pms.PolicyStore, error) {
+
+	var ps pms.PolicyStore
+	services, err := s.ListAllServices()
+	if err != nil {
+		return nil, err
+	}
+	ps.Services = services
+	functions, err := s.ListAllFunctions("")
+	if err != nil {
+		return nil, err
+	}
+	ps.Functions = functions
+	return &ps, nil
+
+}
+
+// WritePolicyStore writes policies to a file
+func (s *Store) WritePolicyStore(ps *pms.PolicyStore) error {
+	err := s.DeleteServices()
+	if err != nil {
+		return err
+	}
+	err = s.DeleteFunctions()
+	if err != nil {
+		return err
+	}
+	for _, service := range ps.Services {
+		err = s.CreateService(service)
+		if err != nil {
+			return err
+		}
+	}
+	for _, f := range ps.Functions {
+		_, err = s.CreateFunction(f)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+
+}
+
+// ListAllServices lists all the services
+func (s *Store) ListAllServices() ([]*pms.Service, error) {
+	serviceCollection := s.client.Database(s.Database).Collection("services")
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	cur, err := serviceCollection.Find(ctx, bson.M{})
+	if err != nil {
+		return nil, err
+	}
+	defer cur.Close(ctx)
+	services := []*pms.Service{}
+	for cur.Next(ctx) {
+		var service pms.Service
+		err := cur.Decode(&service)
+		if err != nil {
+			return nil, err
+		}
+		services = append(services, &service)
+	}
+
+	return services, err
+
+}
+
+// GetServiceNames reads all the service names
+func (s *Store) GetServiceNames() ([]string, error) {
+	serviceCollection := s.client.Database(s.Database).Collection("services")
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	matchstag := bson.D{{"$match", bson.D{{"$exists", true}}}}
+	projectstag := bson.D{{"$project", bson.D{{"_id", 1}}}}
+	opts := options.Aggregate().SetMaxTime(2 * time.Second)
+	cur, err := serviceCollection.Aggregate(ctx, mongo.Pipeline{matchstag, projectstag}, opts)
+	if err != nil {
+		return nil, err
+	}
+	var services []pms.Service
+	if err = cur.All(ctx, &services); err != nil {
+		panic(err)
```

Comment:
Why panic here? panic may terminate the process. Is that possible to return the error?

Reply:
will fix it.